library/pk_wrap.c

```diff
@@ -774,6 +774,8 @@ static int rsa_alt_sign_wrap( void *ctx, mbedtls_md_type_t md_alg,
 #endif /* SIZE_MAX > UINT_MAX */
 
     *sig_len = rsa_alt->key_len_func( rsa_alt->key );
+    if( *sig_len > MBEDTLS_PK_SIGNATURE_MAX_SIZE )
+        return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
```

Comment:
Shouldn't we reset the sig_len in such case?

Reply:
`*sig_len` is unspecified on error. In the mbedtls API, we usually go for minimal code size and leave whatever was there as long as it isn't sensitive.